cuqi/implicitprior/_denoiser.py

```diff
@@ -0,0 +1,108 @@
+#%%
+from abc import ABC, abstractmethod
+from cuqi.distribution import Distribution
+import numpy as np
+
+class DenoiseRegularizer(Distribution):
```

Comment:
I'm trying to understand the role of denoising specifically in the this class, for considering the best possible naming and design.  As far as I can see from the code, this class doesn't actually have anything to do specifically with denoising. What it does seems to be accept some function passed in by the user, then hold it and simply apply it when the `regularize()` method is called.  The function is currently named "denoiser" but it seems it could be other things as well, for example some other filter, like median filter, sobel filter, etc.?   Everything that is actually denoising is done inside this denoising function, which in the demo is represented as a prox_g function.

Reply:
It is a really good point. I think it is related to the naming of the "regularize" method, which in fact does not apply the  implicit regularizer (-log pdf implicit prior), that we cannot compute in principle.
Before this method was called "denoise()" as it consists to perform a denoising task. At this point I would like to emphasize that we have theoretical foundations with denoiser but not Sobel or median filter.
We don't have checks to ensure that the user is providing a denoiser. It is the user responsability at this point.

Reply:
Status update: this discussion refers to the old design.